hack/lib/util/golang.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# This library holds golang related utility functions.
+
+# Ensure the go tool exists and is a viable version.
+os::golang::verify_go_version() {
+  if [[ -z "$(which go)" ]]; then
+    cat <<EOF
+Can't find 'go' in PATH, please fix and retry.
+See http://golang.org/doc/install for installation instructions.
+EOF
+    return 1
+  fi
+
+  local go_version
+  go_version=($(go version))
+  if [[ "${go_version[2]}" < "go1.6" && "${go_version[2]}" != "devel" ]]; then
```

Comment:
Whatever we end up using for the flag, we should have it allow *any* golang version and when doing the version check it should print out a `os::log::warning` that we found a golang version that isn't optimal, but we're going on anyway, and there may be some side-effects